dataactbroker/domainRoutes.py

```diff
@@ -1,14 +1,30 @@
-from dataactbroker.handlers.domainHandler import DomainHandler
+from flask import g
 
+from dataactcore.interfaces.db import GlobalDB
+from dataactcore.models.domainModels import CGAC
+from dataactcore.utils.jsonResponse import JsonResponse
+from dataactcore.utils.statusCode import StatusCode
 
-# Add the file submission route
-def add_domain_routes(app, isLocal, bcrypt):
-    """ Create routes related to domain values for flask app
 
-    """
+# Add the file submission route
+def add_domain_routes(app):
+    """ Create routes related to domain values for flask app """
 
-    @app.route("/v1/list_agencies/", methods = ["GET"])
+    @app.route("/v1/list_agencies/", methods=["GET"])
     def list_agencies():
         """ List all CGAC Agencies """
-        domainHandler = DomainHandler()
-        return domainHandler.list_agencies()
+        sess = GlobalDB.db().session
+
+        if g.user is None:
+            cgacs = []
+        elif g.user.website_admin:
+            cgacs = sess.query(CGAC).all()
+        else:
+            cgac_ids = [affil.cgac_id for affil in g.user.affiliations]
+            cgacs = sess.query(CGAC).filter(CGAC.cgac_code.in_(cgac_ids)).all()
+        agency_list = [
+            {'agency_name': cgac.agencyname, 'cgac_code': cgac.cgac_code}
+            for cgac in cgacs
+        ]
+        return JsonResponse.create(StatusCode.OK,
```

Comment:
There's certainly no new risk in inlining the logic this way (all the same imports are happening and functions being executed in the same order) -- the bigger question is how do we want to break up the logic. In this case, I can move the logic to a "handler" module, but then the routes module would be rather bare:
```python
from dataactbroker.handlers.domain import list_agencies

def add_domain_routes(app):
    app.route("/v1/list_agencies/")(list_agencies)
EOF
```